# Mitch
Mitch is a tool for multi-dimensional enrichment analysis. At it's heart, it uses a rank-MANOVA based statistical approach to detect sets of genes that exhibit enrichment in the multidimensional space as compared to the background. Mitch is useful for pathway analysis of profiling studies with 2 or more contrasts, or in studies with multiple omics profiling, for example proteomic, transcriptomic, epigenomic analysis of the same samples.
## Workflow overview
### Importing profiling data
Mitch accepts pre-ranked data supplied by the user, but also has functions for importing tables generated by Limma, edgeR, DESeq2, ABSSeq and Sleuth. By default, only the genes that are detected in all contrasts are included, but this behaviour can be modified. The below example imports two edgeR tables.
```
x<-list("dge1"=dge1,"dge2"=dge2)
y<-ndrich_import(x,"edger")
```
## Installation
```
install.packages("devtools")
library("devtools")

devtools::install_github("markziemann/Mitch")
library("mitch")
```
